_posts/2021-10-06-#23157.md

```diff
@@ -10,15 +10,80 @@ status: upcoming
 commit:
 ---
 
-_Notes and questions to follow soon!_
+We recently [looked at PR 22677](./22677), which cuts the circular dependency
+between validation and txmempool. This week's PR was split off from that PR,
+taking a slightly different approach to simplify the `CTxMemPool::check()`
+function.
 
-<!-- TODO: Before meeting, add notes and questions
 ## Notes
 
+- The `CTxMemPool` class is a data structure that stores the set of unconfirmed
+  transactions, along with various methods to update or query that set. As well
+  as the transactions themselves, the mempool stores metadata about the
+  transactions (such as ancestor/descendant dependencies) that makes it highly
+  optimized for various operations, such as constructing blocks.
+
+- The mempool must only contain non-conflicting, valid transactions which
+  could theoretically be included in a future block. There are therefore
+  many [invariants](https://en.wikipedia.org/wiki/Class_invariant) such as:
+
+  - a transaction may only appear in the mempool if all of its inputs are
+    either in the UTXO set or are the outputs of other mempool transactions
+  - no two transactions in the mempool may spend the same output
+  - the {mempool|ancestor} {fees|count|size} cached for the transaction must
```

Comment:
I assume this is what it's supposed to say? Or self | descendant | ancestor?
```suggestion
  - the {descendant|ancestor} {fees|count|size} cached for the transaction must
```

Reply:
oops. Thank you!